### ISSUE TITLE
Bound SOUL, Experience, Sleep, consolidation, and model adapters around RelayEngine

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#103]
## Summary

Define the missing **offline-consolidation and persistent-prior boundaries around the candidate RelayEngine** without promoting model adapters, training artifacts, or SLEEP itself into new semantic authorities.

The design question is not whether RelaySelf can fine-tune a model.

It is:

> If cognition is executed by a DiffusionGemma-class RelayEngine, how may Persistent Cognition, SOUL, Experience, Memory, Skill crystallization, and optional model-side adaptation influence future cognition without collapsing source authority into compiled weights?

This Issue is an **architecture / research boundary ledger**. It is not semantic authority, execution authority, or permission for automatic online/self-training.

## Fresh repository state

At creation:

```text
main = bd6440fb1e989ef8e2e988ca8e7b7b625620c62f
tree = 5b3f6cb720ea695dbc516425feb9994aea9911b1
open PRs = 0
ruleset = 23442682 "main protection" (active)
required checks = repository-contracts, pytest, lint
```

Current authority remains primary.

Relevant current lines already exist:

- #101 — DiffusionGemma adaptive cognition candidate;
- #88 — Cognition Is All You Need / Self 1.0 unifying thesis;
- #87 — Skill-centered cognitive narrowing;
- #84 — crystallized / amortized fast cognition;
- #25 — Thinking / Retrieval / Inquiry boundary;
- #46 — bounded viability-gradient physical/model-system evidence;
- #9 — external research literature ledger.

This Issue must reduce to those owners where possible rather than creating parallel concepts.

## Boundary map

The missing surrounding seams can be expressed as:

```text
Persistent Cognition / SOUL
          |
          | project / retrieve / bias
          v
    Present Projection
          |
          v
      RelayEngine
          |
          v
      Skill / Action
          |
          v
      Consequence
          |
          v
      Experience
          |
          +----------------------+
          |                      |
          v                      v
 governed durable          offline consolidation
 integration                   candidate work
          |                      |
          v                      v
 Memory / other          Skill / retrieval /
 Persistent Cognition    adapter candidates
          ^                      |
          |                      v
          +------ validate / accept / reject
```

No arrow in this diagram implies automatic persistence or authority transfer.

## Boundary 1 — SOUL / Persistent Cognition -> Present -> RelayEngine

SOUL and other Persistent Cognition remain source representations.

A candidate execution shape may be:

```text
SOUL / persistent priors
        -> relevant Present projection
        -> current cognition request
        -> RelayEngine
```

Do not require the engine to ingest every durable object on every decision epoch. Retrieval / projection may expose only the relevant subset.

A future compiled representation such as:

```text
SOUL
  -> training examples / preference evidence
  -> candidate soul adapter
```

does **not** invert the authority relation.

Preserve:

```text
SOUL != LoRA
source prior != compiled model-side accelerator
```

Deleting or replacing a compiled adapter must not delete or redefine the source SOUL.

If model behavior conflicts with current authoritative Persistent Cognition, the weights/adapters do not win merely because they are faster.

## Boundary 2 — Present / Focus -> model-side routing

Focus is a present cognitive narrowing, not a required adapter router.

Possible physical implementations may later include:

- ordinary hidden-state conditioning;
- native MoE expert routing;
- prompt/context selection;
- optional LoRA / adapter selection or weighting.

But preserve:

```text
Focus != LoRA router
Skill != LoRA
Appraisal / affect != MoE expert
```

First test whether the base model's own MoE routing and narrowed context already provide useful specialization.

Add explicit adapter routing only when a measured failure demonstrates that it contributes beyond existing model routing and current owners.

## Boundary 3 — Consequence -> Experience

A consequence observation is evidence about what happened.

An Experience is a cognition-facing retained/traceable episode or derived record only when the responsible current design justifies retaining it.

Preserve at minimum:

```text
Consequence evidence
  != Experience
  != durable Memory
  != training example
```

A successful action must not automatically become a positive training target.

A failed action must not automatically become a negative training target.

The interpretation may depend on intent, expected outcome, authority, viability, uncertainty, later evidence, and whether the decision was actually causally responsible for the observed result.

## Boundary 4 — Experience -> Memory / persistent integration

#25 already owns the general question of governed information integration into Persistent Cognition.

This Issue must not duplicate a generic Memory ingestion subsystem.

The narrower question here is:

> Which parts of experience are eligible as consolidation input, and which outputs may propose updates to existing persistent owners?

Candidate consolidation may discover:

- recurring retrieval structure;
- recurring Skill-local context;
- repeated decision patterns;
- candidate abstractions;
- candidate Skill refinements;
- candidate training / adapter examples.

Those remain proposals/evidence until accepted by the responsible persistent owner or authority path.

## Boundary 5 — Experience -> training material

Training material is a **derived dataset**, not raw truth.

A candidate pipeline may be:

```text
Experience / trace
  -> eligibility filtering
  -> causal / outcome interpretation
  -> deduplication / balancing
  -> candidate example
  -> held-out split
  -> training candidate
```

This derivation should retain provenance back to the source experience/evidence where practical.

Do not silently train on:

- generated self-claims as if they were attested facts;
- model outputs merely because they were fluent;
- outcomes whose causal relation to the decision is unknown;
- stale/retracted persistent cognition;
- unreviewed authority-sensitive instructions;
- a single repeated episode merely because it dominates the local log.

## Boundary 6 — SLEEP -> offline cognition allocation

SLEEP should remain reducible, if possible, to a state / Skill / scheduling condition that reallocates cognition away from external interaction toward internal work.

Candidate shape:

```text
SLEEP / offline interval
  -> reduce external cognition load
  -> replay / consolidate / evaluate
  -> produce candidates
  -> validate
  -> governed integration
```

Preserve:

```text
SLEEP != Memory owner
SLEEP != learning authority
SLEEP != automatic weight mutation
```

The same consolidation pipeline could in principle run during another quiet/offline period. Literal biological sleep semantics are not required.

## Boundary 7 — Consolidation -> Skill crystallization

#84 and #87 already own the general crystallization and Skill-narrowing hypotheses.

This Issue only defines the seam:

```text
repeated useful cognition
  -> candidate reusable structure
  -> held-out / consequence validation
  -> accepted Skill / local typed mechanism if justified
```

Human-readable naming can describe a candidate but cannot validate it.

Do not create a generic Skill registry merely because consolidation generates candidate Skills.

## Boundary 8 — Consolidation -> model adapter

A model-side adapter is an optional compiled execution artifact.

Possible research examples include:

- decision-local LoRA;
- Skill-local adapter;
- stable personality / SOUL-derived adapter;
- SFT or preference-tuned adapter;
- other bounded parameter-efficient modifications.

The key test is whether the adapter makes useful cognition cheaper or more reliable without violating current semantics.

A candidate may be evaluated through:

```text
same held-out decision class

before:
  outcome quality
  denoising steps
  latency
  failure / escalation behavior

after candidate adapter:
  same measurements
  + regression tests
  + authority / viability checks
```

A lower training loss is not sufficient evidence.

## Boundary 9 — candidate adapter -> accepted runtime artifact

Do not automatically activate a newly trained artifact.

Before any future permanent mechanism is proposed, the experiment should distinguish at least conceptually:

```text
candidate
  -> validated
  -> usable / selected for an experiment
  -> rejected / superseded
```

This does not yet justify a repository-wide adapter registry or lifecycle owner.

The minimum representation should remain experiment-local until repeated consumers demonstrate irreducible shared requirements.

Required validation may include:

- held-out task quality;
- old-behavior regression;
- novel-state escalation behavior;
- false-confidence / premature-stop rate;
- SOUL / Persistent Cognition consistency where relevant;
- viability / authority boundary preservation;
- rollback to the unmodified base model.

## Boundary 10 — accepted compiled artifact -> RelayEngine

Compiled artifacts may bias execution but must remain replaceable.

A candidate runtime may conceptually be:

```text
authoritative/relevant Present inputs
        +
base DiffusionGemma
        +
optional validated adapter(s)
        |
        v
RelayEngine cognition
```

The engine result remains cognition output, not authority.

If the adapter is absent, incompatible, rejected, or stale, the system should be able to fall back to the base cognition path rather than losing the underlying persistent Self.

## Crystallization metric hypothesis

One useful measurable signature of consolidation may be:

```text
same or better useful outcome
+
fewer denoising steps / less context / less latency / less compute
```

For example:

```text
novel:
  12 steps

experienced:
  5 steps

consolidated:
  2 steps
```

This is a candidate **cost-reduction observation**, not a definition of learning.

Learning may also improve quality, calibration, retrieval, robustness, or recovery rather than only reducing step count.

## Reversibility requirement

Compression must not become a trap.

A crystallized path must remain penetrable by contradiction, changed Present state, new evidence, viability pressure, authority changes, or failed consequences.

Conceptually:

```text
fast compiled path
  -> mismatch / uncertainty / contradiction
  -> disable or bypass shortcut
  -> broaden cognition / retrieve / inquire
  -> deeper RelayEngine cognition
```

This reversibility is more important than forcing every decision into the cheapest path.

## Grand Null

Do **not** create merely to support this design:

- a global `ConsolidationState`;
- a new semantic `Learning` owner;
- a repository-wide adapter registry;
- a permanent `SleepManager`;
- a `SoulLoRA` authority surface;
- an `ExperienceDatabase` before current Memory/evidence owners are shown insufficient;
- automatic adapter selection as a semantic Focus owner;
- automatic nightly fine-tuning;
- direct persistence of model-generated interpretations as facts;
- a generic Skill schema.

Add a shared owner/interface only if materially different concrete consumers repeatedly require the same non-reducible state, authority, or lifecycle.

## First bounded example

Do not start by training a personality adapter.

First use one narrow decision class where #101 can already measure cognition cost.

Example:

```text
same bounded decision family
same base DiffusionGemma
same evaluation set

A: base model
B: base model after experiment-local consolidation adapter
```

Train only on a disjoint experience set.

Measure whether B:

- preserves or improves held-out outcome quality;
- reduces required denoising steps or latency;
- still escalates on out-of-distribution / misleading-focus cases;
- does not create new authority/viability violations;
- can be disabled with clean fallback to A.

Only after this works should SOUL-derived personality adapters or Focus-dependent adapter selection receive their own bounded experiment.

## Success criterion

This Issue succeeds when RelaySelf has a minimal, non-duplicative explanation for:

1. how SOUL / Persistent Cognition influence RelayEngine without becoming weights;
2. how consequence evidence may become Experience without automatic persistence;
3. how Experience may feed offline consolidation without automatically becoming Memory or training truth;
4. how SLEEP can allocate cognition without owning learning;
5. how Skill / adapter candidates remain proposals until validated;
6. how a compiled model artifact may accelerate cognition without replacing source authority;
7. how failed crystallization can be bypassed or rolled back.

No code or new owner is required merely to complete the conceptual reduction.

Refs #101.
Refs #88.
Refs #87.
Refs #84.
Refs #25.
Refs #46.
Refs #9.

**Comment by rinsakamo** [comment]
Closed-loop follow-up: #104 defines the mismatch signal that may later feed Experience/consolidation. It explicitly prevents direct failure-to-training conversion and requires causal/provenance distinctions before #103-style consolidation can use the episode.

**Comment by rinsakamo** [comment]
Completed by PR #119 and promoted into current runtime authority.

Terminal reduction:

```text
Persistent Cognition
  -> remains source authority

consequence / trace
  -> candidate consolidation material
  != automatic Memory
  != training truth

offline / sleep interval
  -> cognition-allocation condition
  != learning authority

Skill / policy / procedure / adapter candidate
  -> validation before use

compiled cognition
  -> replaceable accelerator
  != Persistent Cognition
  != new semantic authority

mismatch / novelty / uncertainty / authority change
  -> bypass compiled shortcut
  -> broaden cognition / retrieve / inquire
```

No `ConsolidationState`, `Learning` owner, `SleepManager`, adapter registry, automatic training loop, or generic Skill schema was introduced.

Merged main: `86955cd15b6a349b331a5882570b7ed7f9735058`
Tree: `e88b2fd29025d581aac360b32be602aa46204a85`
PR head: `2cbc8a379ba5132c8859696be9fa301cb3aadbcb`
Required checks: repository-contracts / pytest / lint all success.

Empirical claims about whether a particular adapter or consolidation mechanism improves cognition remain with later bounded experiments; they are not part of #103's completed conceptual reduction.
